src/store/todo/todoSagas.ts

```diff
@@ -1,19 +1,24 @@
 import { getTodosApi } from "api/todoApi";
 
 import { TodoType } from "constants/types/TodoTypes";
-import { call, fork, put, takeLatest } from "redux-saga/effects";
+import { call, fork, put, takeLatest, delay } from "redux-saga/effects";
 
 import * as todosActions from "./todoActions";
 import * as todosTypes from "./todoTypes";
 
-function* getTodoSagas() {
+function* getTodoSagas(action: todosTypes.FetchTodos) {
+  const { payload: callback } = action;
   try {
     yield put(todosActions.fetchTodosPending());
     const todos: TodoType[] = yield call(getTodosApi);
+    callback("Pobieranie w toku", "SHOW");
+    yield delay(2000);
```

Comment:
I suggested you to use delay in non-async actions like Add, Delete, Edit to simulate async